internal/server/everest.go

```diff
@@ -152,19 +154,34 @@ func (e *EverestServer) initHTTPServer(ctx context.Context) error {
 	}
 	apiGroup.Use(jwtMW)
 
-	// Setup and use RBAC (casbin) middleware.
-	rbacMW, err := e.rbacMiddleware(ctx, basePath)
-	if err != nil {
-		return err
-	}
-	apiGroup.Use(rbacMW)
-
 	apiGroup.Use(e.checkOperatorUpgradeState)
-	RegisterHandlers(apiGroup, e)
+	api.RegisterHandlers(apiGroup, e)
 
 	return nil
 }
 
+//nolint:ireturn
+func newHandlerChain(
+	ctx context.Context,
+	log *zap.SugaredLogger,
+	kubeClient *kubernetes.Kubernetes,
+	vsURL string,
+) (handlers.Handler, error) {
+	// Initialise handlers.
+	k8s := k8shandler.New(log, kubeClient, vsURL)
+	rbacH, err := rbachandler.New(ctx, log, kubeClient)
+	if err != nil {
+		return nil, errors.Join(err, errors.New("could not create rbac handler"))
+	}
+	validation := validation.New(log, kubeClient)
+
+	// Chain handlers.
+	// validation -> rbac -> k8s
+	validation.SetNext(rbacH)
+	rbacH.SetNext(k8s)
+	return validation, nil
```

Comment:
Don't know if it was intentional or not but you resolved this conversation without implementing the naming changes of `k8s`->`k8sH` and `validation`->`valH`. It's fine if you don't agree with the suggestion but since you reacted with :+1: I thought it might have been a mistake to resolve the conversation.

Reply:
I incorrectly marked it as resolved because it went "outdated" before I could get to it. Sorry about that, its fixed now